sympy/integrals/singularityfunctions.py

```diff
@@ -55,7 +55,7 @@ def singularityintegrate(f, x):
         n = sympify(f.args[2])
         if n.is_positive or n.is_zero:
             return SingularityFunction(x, a, n + 1)/(n + 1)
-        elif n in (-1, -2):
+        elif n in (S.NegativeOne, S(-2)):
```

Comment:
In general it is better to use `Integer(2)` rather than `S(-2)` to avoid going through sympify. Otherwise for clarity `-1, -2` is fine but it does imply sympification.

Reply:
I think `n in (-1, -2)`. is better. If that `x == 1` slow because `x` has to sympify the rhs we could improve the performance of that by improving `__eq__`.

Reply:
For instance, `Basic.__eq__` could short-circuit False on `int`, which would be overridden by `Integer.__eq__`. I expect optimizing equality checks in this way could lead to significant performance gains throughout the library, as == checks are done everywhere (any time an object is in a container like a dict or a list it is going to have a lot of == checks done on it). Right now the very first thing `Basic.__eq__` (and `Expr.__eq__`) does is call sympify, but actually constructing a sympified object can be avoided for almost every sympifyable type.

Reply:
Not sure I fully get the `__eq__` aspects. I've just noted that there is a tremendous amount of calls to `sympify` in almost every benchmark I've tested.

`Integer(-2)` is indeed better. (Except for readability...)

Reply:
Another aspect of performance here is that `(-1, -2)` can be generated at compile time. Even though `(S.NegativeOne, Integer(-2))` are constants, it may be that they cannot.

(I know this is just a very small part of a not often used function, it is more the general concept of trying to write code in an efficient way wherever possible that I am looking for...)

Reply:
Again, readability should be the forefront. Microbenchmarks are a waste of time, and usually end up making code harder to maintain. They should only be considered for code that gets run millions of times because it is deep in the core.